internal/broker/bind.go

```diff
@@ -3,36 +3,328 @@ package broker
 import (
 	"context"
 	"fmt"
+	"net/http"
+	"sync"
 
+	jsonhash "github.com/komkom/go-jsonhash"
 	"github.com/pkg/errors"
 	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	"github.com/sirupsen/logrus"
+	"k8s.io/helm/pkg/proto/hapi/chart"
 
 	"github.com/kyma-project/helm-broker/internal"
 )
 
 type bindService struct {
-	instanceBindDataGetter instanceBindDataGetter
+	addonIDGetter                 addonIDGetter
+	chartGetter                   chartGetter
+	instanceGetter                instanceGetter
+	bindTemplateRenderer          bindTemplateRenderer
+	bindTemplateResolver          bindTemplateResolver
+	instanceBindDataInserter      instanceBindDataInserter
+	instanceBindDataGetter        instanceBindDataGetter
+	instanceBindDataRemover       instanceBindDataRemover
+	bindStateGetter               bindStateBindingGetter
+	bindOperationGetter           bindOperationGetter
+	bindOperationCollectionGetter bindOperationCollectionGetter
+	bindOperationInserter         bindOperationInserter
+	bindOperationUpdater          bindOperationUpdater
+	operationIDProvider           func() (internal.OperationID, error)
+	mu                            sync.Mutex
+
+	log *logrus.Entry
+
+	testHookAsyncCalled func(internal.OperationID)
 }
 
-func (svc *bindService) Bind(ctx context.Context, osbCtx OsbContext, req *osb.BindRequest) (*osb.BindResponse, error) {
+func (svc *bindService) Bind(ctx context.Context, osbCtx OsbContext, req *osb.BindRequest) (*osb.BindResponse, *osb.HTTPStatusCodeError) {
```

Comment:
The service layer is not HTTP protocol aware

Reply:
Provision service is implemented the same way - separating HTTP protocol layer from services requires more custom errors which would be handled in server layer.